dao/src/main/java/org/wso2/testgrid/dao/repository/TestPlanRepository.java

```diff
@@ -133,4 +134,91 @@ public TestPlan findByPrimaryKey(String id) throws TestGridDAOException {
                     " query [", queryStr, "]"), e);
         }
     }
+
+    /**
+     * This method returns the last failed {@link TestPlan} for a given product.
+     *
+     * @param product A Product object being queried.
+     * @return instance of a {@link TestPlan} representing the last failed test plan.
+     * @throws TestGridDAOException when an error occurs while executing the native sql query.
+     */
+    public TestPlan getLastFailure(Product product) throws TestGridDAOException {
+        String sql = "SELECT  t.* from test_plan t INNER JOIN (SELECT tp.infra_parameters," +
```

Comment:
Shall we use `PrepardStatement`s instead?

Reply:
Used parameters instead preparedStatements